runtime/common/src/zenlink.rs

```diff
@@ -0,0 +1,257 @@
+use spacewalk_primitives::{Asset, CurrencyId};
+
+use zenlink_protocol::{LOCAL, NATIVE};
+pub type ZenlinkAssetId = zenlink_protocol::AssetId;
+
+pub const USDC_ISSUER: [u8; 32] = [
+	59, 153, 17, 56, 14, 254, 152, 139, 160, 168, 144, 14, 177, 207, 228, 79, 54, 111, 125, 190,
+	148, 107, 237, 7, 114, 64, 247, 246, 36, 223, 21, 197,
+];
+
+pub const BRL_ISSUER: [u8; 32] = [
+	234, 172, 104, 212, 208, 227, 123, 76, 36, 194, 83, 105, 22, 232, 48, 115, 95, 3, 45, 13, 107,
+	42, 28, 143, 202, 59, 197, 162, 94, 8, 62, 58,
+];
+
+pub const TZS_ISSUER: [u8; 32] = [
+	52, 201, 75, 42, 75, 169, 232, 181, 123, 34, 84, 125, 203, 179, 15, 68, 60, 76, 176, 45, 163,
+	130, 154, 137, 170, 27, 212, 120, 14, 68, 102, 186,
+];
+
+fn discriminant(currency: &CurrencyId) -> u8 {
+	match currency {
+		CurrencyId::Native => 0,
+		CurrencyId::XCM(_) => 1,
+		CurrencyId::Stellar(_) => 2,
+		CurrencyId::ZenlinkLPToken(_, _, _, _) => 6,
+	}
+}
+
+pub fn zenlink_id_to_currency_id(asset_id: ZenlinkAssetId) -> Result<CurrencyId, ()> {
```

Comment:
I thought about how we use this function and what it should do.

I realized that it is probably a good idea to check that the `chain_id` contained in the `asset_id` passed to this function matches the ID of our current parachain. At the moment, we do not check the `chain_id` in this function which is problematic.

Let's say pass an `asset_id` for a Bifrost asset here. The `asset_index` conversion logic we implement is roughly the same so we might actually end up treating Bifrost assets like our own. 

Anyways, I think it might be a good idea to add a simple check for the `chain_id` to this function. Or am I exaggerating here and this would never be a problem @adelarja? What do you think?

Reply:
@ebma I think it is a good idea.

I'll change the function to receive the parachain_id as a parameter and we can call  `ParachainInfo::parachain_id()` in each parachain runtime configuration when we call the function.

The function signature will change to something like:

```rust
pub fn zenlink_id_to_currency_id(asset_id: ZenlinkAssetId, parachain_id: u32) -> Result<CurrencyId, ()>
```

and we will call the function doing something similar to:

```rust
if let Ok(currency_id) = zenlink_id_to_currency_id(asset_id,  ParachainInfo::parachain_id())
```

What do you think?

Reply:
Sounds good 👍